integration/test/roster_test.js

```diff
@@ -0,0 +1,285 @@
+const { describe, before, after, it } = require('mocha');
+const { v4: uuidv4 } = require('uuid');
+const RosterTestPage = require('../pages/RosterTestPage');
+const DriverFactory = require('../utils/DriverFactory');
+
+describe('Roster Test', async function () {
+  const chromeDriverFactory = new DriverFactory();
+  const firefoxDriverFactory = new DriverFactory();
+  let localPage;
+  let remotePage;
+  const meetingId = uuidv4();
+  localAttendeeName = 'local-attendee';
+  remoteAttendeeName = 'remote-attendee';
+  let failureCount = 0;
+
+  before(async function () {
+    await chromeDriverFactory.build('Roster Test - Local', 'sauce-chrome');
+    localPage = new RosterTestPage(chromeDriverFactory.driver);
+
+    await firefoxDriverFactory.build('Roster Test - Remote', 'sauce-firefox');
+    remotePage = new RosterTestPage(firefoxDriverFactory.driver);
+  });
+
+  afterEach(function () {
+    const state = this.currentTest.state;
+    if (state === "failed") {
+      failureCount += 1;
+    }
+  });
+
+  after(async function () {
+    const passed = failureCount === 0;
+    await chromeDriverFactory.quit(passed);
+    await firefoxDriverFactory.quit(passed);
+  });
+
+  describe('Load App', async function () {
+    it('chrome', async function () {
+      await localPage.load('/roster-test');
+      await localPage.checkIfAppLoaded();
+    });
+
+    it('firefox', async function () {
+      await remotePage.load('/roster-test');
+      await remotePage.checkIfAppLoaded();
+    });
+  });
+
+  describe('Join Meeting', async function () {
+    it('local attendee', async function () {
+      await localPage.enterMeetingInfo(meetingId, localAttendeeName);
+      await localPage.joinMeeting();
+      await localPage.checkIfAttendeeHasJoinedMeeting();
+    });
+
+    it('remote attendee', async function () {
+      await remotePage.enterMeetingInfo(meetingId, remoteAttendeeName);
+      await remotePage.joinMeeting();
+      await remotePage.checkIfAttendeeHasJoinedMeeting();
+    });
+  });
+
+  describe('Attendee Presence', async function () {
+    describe('#length', async function () {
+      describe('component', async function () {
+        it('has length of 2', async function () {
+          await localPage.checkRosterLength(2);
+        });
+      });
+
+      describe('hook', async function () {
```

Comment:
Checking if can we use the actual name of the hook in place of 'hook'?

Reply:
Updated.